subprojects/android/src/main/java/org/mockito/android/internal/creation/AndroidByteBuddyMockMaker.java

```diff
@@ -1,13 +1,32 @@
 package org.mockito.android.internal.creation;
 
 import org.mockito.internal.creation.bytebuddy.SubclassByteBuddyMockMaker;
+import org.mockito.internal.util.ConsoleMockitoLogger;
+import org.mockito.internal.util.Platform;
 import org.mockito.invocation.MockHandler;
 import org.mockito.mock.MockCreationSettings;
 import org.mockito.plugins.MockMaker;
 
-public class AndroidByteBuddyMockMaker implements MockMaker {
+import static org.mockito.internal.util.StringJoiner.join;
 
-    private final MockMaker delegate = new SubclassByteBuddyMockMaker(new AndroidLoadingStrategy());
+public class AndroidByteBuddyMockMaker extends ConsoleMockitoLogger implements MockMaker {
+
+    private final MockMaker delegate;
+
+    public AndroidByteBuddyMockMaker() {
+        if (Platform.isAndroid() || Boolean.getBoolean("org.mockito.mock.android")) {
+            delegate = new SubclassByteBuddyMockMaker(new AndroidLoadingStrategy());
+        } else {
+            log(join(
```

Comment:
I would not want to tear the two apart. As it is the only class and the constructor is only a few lines, I do not think that this would improve readability as this would only mean that the same logic was moved to a method.